### DISCUSSION TITLE
✅ 🌟 Introducing Elementor AI-generated containers 🌟

### DISCUSSION THREAD

**Discussion by avivu** [elementor/elementor#24352] [Announcement]
![AI Text to Elementor](https://github.com/elementor/elementor/assets/76733085/7235f14a-be14-4659-b48b-fab26a365558)

**Hey Beta testers,** 👋

New AI capabilities are available in Beta 3.18.1 and we're thrilled for you to try them out and provide you feedback. At Elementor, we understand the importance of empowering you as our user, providing tools to enhance your workflows process and expand your creative toolbox for ideation, layout setup, and content generation.
We would like to introduce you to AI generated containers. Now you can effortlessly construct containers with any Elementor widget at your disposal.
We offer you three distinct levels for generating containers with AI:

1. **Text-Based Container Generation:** Create on-demand designed containers by using a single textual prompt.

2. **Elementor-Based Container Variations:** Edit containers based on existing Elementor layouts and customizing their content, images, fonts, colors, and icons.

3. **Web-Based Container generation:** Create containers by providing a specified URL to generate Elementor-compatible layouts. Content is then refined to reflect your prompt’s request while preserving the original structure.

We encourage you to try out this beta version and provide us with your thoughts on our latest additions.

&nbsp;

# Bring Your Vision to Life Effortlessly with Text-Based Container Generation

This version includes a phenomenal leap forward in Elementor’s journey to redefine the web creation experience. In previous versions, Elementor AI allowed you to generate and edit text, code, and images within Elementor widgets. Now, Elementor AI empowers you to generate entire container layouts and content from scratch with a prompt saving you an impressive amount of time, improving your workflow and boosting your creativity. Just enter a description of the section you want and Elementor AI will build the layout and populate it with content. Re-generate results to get more possible variations to your request.

Use our prompt enhancer to write better prompts. It will automatically add suggested details to your written prompts to improve your generated results.
 

#### How to Test It:

1. Before starting, head to your WordPress dashboard, click Elementor > Settings > Features and activate the “Build with AI” experiment. Please note that this feature requires the Containers feature to be active on your site.
2. Create a new page and edit it using Elementor
3. On your canvas, you will now find the "Build with AI" button next to the "Add new container" and "Library" buttons. This button allows you to input your prompt and receive three results based on it.
4. Add your prompt, including the layout, style, elements, and the topic of the section you would like to generate. You can click “/” to open the pre-made suggestions that will assist you through the prompt creation process.
5. If you don’t know how to write your request, click on the magic wand icon to enhance your prompt.
6. Click "Generate" and wait for 10-45 seconds for the AI to generate your three options.
7. If you don't like the results, you can always regenerate three more options and choose the template that suits you best.
8. If you like a result, go ahead and insert it into your page. From this point, you can edit and adjust it to meet your needs, such as replacing images, modifying content, adjusting colors, typography, and more.
9. Try to create an entire page using this feature, section by section, and make sure the results suit your requirements
10. Important Note: When testing on a website that doesn't have globally assigned fonts and colors (aside from the default ones), the AI will generate a palette and fonts based on your prompt. If you have a global design system on your site, it will attempt to use it as much as possible.

&nbsp;

# Elevate Your Design Concepts with Elementor-Based Container Generation

Elementor-Based Container Variations provide the means to boost your website's design with ease. By utilizing existing Elementor layouts as a foundation, you can craft variations that perfectly align with your unique design concepts. This AI capability lets you keep the container’s layout and widgets while customizing various elements, including content, images, fonts, colors, and even icons. The result is a super fast creation workflow with a design that reflects your individual style without the need to rebuild and adjust the design manually. Use templates you’ve already built and wish to recreate on a new website, conduct a/b testing between design easily, and much more. Just choose a container, right-click and choose “Edit with AI”, add a prompt to indicate your design adjustments specifications, and get three new design variations.

#### How to Test It:

1. Create a new page or edit an existing one using Elementor.
2. You can either insert a new template from the library, choose one of the AI-generated templates, or use your self-created templates with assigned styling values.
3. Right-click on the container you want to generate variations for and select "Edit with AI."
4. Ensure that the thumbnail of your selected container is available and attached, and then add your request for changes. You can describe any changes to your content, colors, typography, images, and icons. If you need assistance in formulating your prompt, you can click on the "Enhance variation" button to have your original prompt enhanced by AI.
5. Click "Generate" and review the three options you receive based on your original container.
6. Insert your selected variation onto your page and repeat this process for additional containers if needed.
7. Important Note: When testing on a website that doesn't have globally assigned fonts and colors (aside from the default ones), the AI will generate a palette and fonts based on your prompt. If you have a global design system on your site, it will attempt to use it as much as possible.

&nbsp;

# [Coming Soon] Advanced AI at Your Fingertips with Web-Based Container Generation

Level up with the advanced and innovative approach of Web-Based Containers, offering an extraordinary solution. Provide a specific URL, select the reference-based section, and witness the magic of Elementor AI in action. This advanced feature opens up the entire web as your reference, saving time and streamlining your workflow for effortless adjustments. The results will deliver generative Elementor-compatible layouts seamlessly integrated into your website. By using a prompt to define your exact requests, AI refines the content and provides three variations, all while preserving the original structure. Redefine your web creation experience, seamlessly adapting to the ever-evolving digital world.

#### How to Test It:

1. Create a new page and edit it using Elementor.
2. On your canvas, you will now find the "Build with AI" button located next to the "Add new container" and "Library" buttons. Click on it to launch the prompt modal.
3. Click on the “+” button next to the text input field and select “URL as reference” from the dropdown menu.
4. Insert the URL of a website you would like to draw reference from and click “Submit”.
5. Select the block from the website that you want the AI to take inspiration from and click “Apply”. Make sure the selected reference is visible in the attachment thumbnail.
6. Add a prompt and click “Generate”.
7. Choose the most suitable option from the three variations generated to meet your needs.
8. Important Notice: Currently, this feature is still in development, and we will provide updates once it's released and fully operational. Please be aware that this feature is not yet fully optimized at the moment and may require minor adjustments for responsiveness after being inserted.

&nbsp;

# AI image Improvements

We are thrilled to announce that we have recently made significant updates to our image generation AI models, resulting in substantial improvements in terms of image quality, accuracy, and variety. Using our tools to create images with AI you can now expect even better results that will elevate your creativity. These improvements represent our ongoing commitment to delivering cutting-edge AI solutions and ensuring that your experience with our services continues to exceed your expectations. Explore our updated tools and experience firsthand these remarkable improvements, bringing your ideas to life with AI-generated images like never before.

-----

## Make An Impact: Share Feedback Before Features Are Released

We’re always looking to get to know Elementor customers on a deeper level. If you’re interested in meeting with our team and sharing your insights, feedback, and feature requests, you’re welcome to sign up for some quality time with us [here](https://elementor.com/pages/joinresearch/).

### How to Install the Beta Version
If you still haven't enabled beta testing in Elementor, please follow these steps:

- Go to Elementor > Tools > Versions
- Enable the beta tester feature
- Go to Dashboard > Updates and update to the latest beta version.
- This will set you up to test the new Elementor Free and Pro 3.18 Beta.

----

## Notes

As we make changes to our GitHub in an effort to keep it organized, improve the way we deliver information to you, and our ability to review your feedback, please follow the instructions below when reporting bugs, or making feature requests:

- To report a bug related to beta versions - open a new [Bug Report](https://github.com/elementor/elementor/issues/new?assignees=&labels=status%2Fawaiting_triage&template=bug.yml)
- For feature requests - open a new [Feature Request](https://github.com/orgs/elementor/discussions/new?category=feature-request)
- For other comments specifically about this beta version - comment here

Designated time for release: `December 4, 2023 (Estimated)`

Many thanks for your support and help!

Cheers,
_Elementor Team_

**Comment by pauserratgutierrez** [comment]
Do we get free tokens to try out the beta or do they count on the total we have?

**Reply by shilo-ey** [reply]
You can use it regardless of the tokens you own, currently they won’t count in your sum of tokens.

**Reply by fabigrafo** [reply]
But only if we have Elementor AI, right?

**Reply by shilo-ey** [reply]
You can use this with a free subscription as well

**Reply by perks98** [reply]
@shilo-ey it doesn't work for me. My trial ended a long long time ago so I'm guessing if the token count is zero you can't test this beta?

![Screenshot 2023-11-21 085453](https://github.com/elementor/elementor/assets/55829113/86b35590-2482-4552-8f62-378793801164)

**Reply by shilo-ey** [reply]
@perks98 please try to create a new account and retry. you need an account with standing amount of tokens to use it.

**Comment by m1ga** [comment]
Have to say that the results are very random. In the best case one of the three results is somehow matching but I get better results by just using a premade block or just create a layout myself. 

![Screenshot_20231121_130845](https://github.com/elementor/elementor/assets/4334997/2bbe7d5a-2664-4ced-9846-5a293237b558)

Yes, it is a very basic prompt but the results are not very useful. It's looking like it is randomly returning presets. I even got the first result for a different prompt before!

**Reply by nicholaszein** [reply]
Hey @m1ga,

as with all things AI, it gets better exponentially as time passes. We hope that with your feedback, we can adjust and improve the results to better fit our users' needs. However, we do expect it to get better as more users try it and use it. So thank you for your feedback! 🙏

**Comment by shilo-ey** [comment]
As we continue to enhance and refine AI Generated Containers feature, your feedback is invaluable to us. 
We're excited to announce that we are conducting user interviews to gather in-depth insights and suggestions directly from you, our users!

**📅 Schedule Your Interview:** We've made it easy for you to book a time slot that works for you. Please visit our scheduling [link here](https://calendly.com/shilo-ey/30min) to book a 30-minute interview session.

**🔍 What to Expect:** During the interview, we'll discuss your experiences with our AI product, any challenges you've faced, and any improvements or features you'd like to see. Your honest feedback will directly influence our development roadmap!

📝 **Your Privacy Matters**: All interviews will be confidential, and your feedback will be used solely for the purpose of improving the product based on your feedback.

We're looking forward to having meaningful conversations with you and making [Your Product/Project Name] even better with your help.

Thank you for being a part of our journey!

Best,
Shilo from Elementor
